src/bulk/common.ts

```diff
@@ -660,16 +665,39 @@ function handleMongoWriteConcernError(
  * @public
  * @category Error
  */
-export class BulkWriteError extends MongoError {
+export class MongoBulkWriteError extends MongoError {
   result?: BulkWriteResult;
 
-  /** Creates a new BulkWriteError */
+  /** Number of documents inserted. */
+  insertedCount?: number;
```

Comment:
If `result` isn't optional (see lower comment) none of these new properties need to be optional either.